src/Appwrite/Platform/Modules/Functions/Workers/Builds.php

```diff
@@ -705,6 +708,33 @@ protected function buildDeployment(Device $deviceForFunctions, Device $deviceFor
                 throw new \Exception('Build size should be less than ' . number_format($buildSizeLimit / 1048576, 2) . ' MBs.');
             }
 
+            if ($resource->getCollection() === 'sites' && empty($resource->getAttribute('adapter'))) {
```

Comment:
Instead of checking for sites or functions, we should have feature flags for the build worker that we can pass true or false from the API when triggering the job to control how we use them. Each worker, API container should be designed as general purpose service for more flexible design. In this case the job should have a boolean attribute called `detection`.

Reply:
Do you agree to ignore it in this PR and have task for next week? I would like to separate feature and code-quality, so we can do QA sooner

Reply:
(plus this is already problem in many many places of worker, and Appwrite router)

Reply:
Will be taken care of in a separate PR